store/postgres/src/relational_queries.rs

```diff
@@ -1646,25 +1646,53 @@ impl<'a> FilterCollection<'a> {
 pub struct SortKey<'a> {
     column: Option<&'a Column>,
     value: Option<&'a str>,
-    direction: EntityOrder,
+    direction: &'static str,
+    /// Entirely skip creating an `order by` clause. The other fields of this
+    /// struct will be ignored if this is `true`
+    omit: bool,
 }
 
 impl<'a> SortKey<'a> {
     /// Generate selecting the sort key if it is needed
     fn select(&self, out: &mut AstPass<Pg>) -> QueryResult<()> {
-        if let Some(column) = self.column {
-            let name = column.name.as_str();
-            if !column.is_primary_key() {
-                out.push_sql(", c.");
-                out.push_identifier(name)?;
+        if !self.omit {
+            if let Some(column) = self.column {
+                let name = column.name.as_str();
+                if !column.is_primary_key() {
+                    out.push_sql(", c.");
+                    out.push_identifier(name)?;
+                }
             }
         }
         Ok(())
     }
 
     /// Generate
-    ///   order by [name direction,] id
+    ///   order by [name direction], id
     fn order_by(&self, out: &mut AstPass<Pg>) -> QueryResult<()> {
+        if self.omit {
+            Ok(())
+        } else {
+            out.push_sql("order by ");
+            self.sort_expr(out)
+        }
+    }
+
+    /// Generate
+    ///   order by g$parent_id, [name direction], id
+    fn order_by_parent(&self, out: &mut AstPass<Pg>) -> QueryResult<()> {
+        if self.omit {
+            out.push_sql("order by g$parent_id ");
```

Comment:
This is confusing, not that I understand much about the window queries, but why is ordering by the parent id still important even `omit` was specified?

Reply:
Because it's only done for the outer part of queries where we already have collected actual children. I should probably check, but at least in some incarnation of `prefetch.rs`, the join logic expected children to come back sorted by parent, and that's another aspect of this logic that I didn't want to touch right now.

Reply:
I just looked at the code that does the [join](https://github.com/graphprotocol/graph-node/blob/master/graphql/src/store/prefetch.rs#L372-L385) and it does not require that children are sorted by parent id. They do need to appear in the requested order for each parent, but in the `Unordered` case that doesn't matter. I'll change `order_by_parent`

Reply:
So in that case, `omit` is true even though the query is expected to return more than one row? And this makes sense because we are re-ordering the children which are already in the correct relative order for each parent?